hawkular-alerts-engine/src/main/java/org/hawkular/alerts/engine/impl/AlertsEngineImpl.java

```diff
@@ -762,19 +773,21 @@ private void handleAutoResolvedTriggers() {
 
     private int checkMissingStates() {
         int numMatchingEvals = 0;
-        for (MissingState missingState : missingStates) {
+        Iterator<MissingState> itMissingStates = missingStates.iterator();
```

Comment:
Because missingStates can be updated concurrently by other thread.
So using a for is not safe on this case.